src/common/Collision/Models/WorldModel.cpp

```diff
@@ -343,7 +348,7 @@ namespace VMAP
         delete iLiquid;
         iLiquid = nullptr;
 
-        if (result && fread(&iBound, sizeof(G3D::AABox), 1, rf) != 1) { result = false; }
+        if (fread(&iBound, sizeof(G3D::AABox), 1, rf) != 1) { result = false; }
```

Comment:
Does C++ not do exceptions? Looks like those would make this code a lot more readable

Reply:
@steenburgh not sure if I got what you meant, but basically `result` is always truthy there

Reply:
Nah that part earlier makes sense. I'm just commenting on the code style in general

```
result = true
if(doThing1() != 1) { result = false }
if(result && doThing2() != 1) { result = false }
if(result && doThing3() != 1) { result = false }
...
return result
```

... is a lot uglier than
```
try {
    doThing1()
    doThing2()
    doThing3()
    return true;
} catch () {
    return false;
}
```